engine/storage/src/org/apache/cloudstack/storage/datastore/ObjectInDataStoreManagerImpl.java

```diff
@@ -172,7 +175,11 @@ public DataObject create(DataObject obj, DataStore dataStore) {
                     if (snapshotDataStoreVO != null) {
                         ss.setParentSnapshotId(snapshotDataStoreVO.getSnapshotId());
                     }
-                    ss.setInstallPath(TemplateConstants.DEFAULT_SNAPSHOT_ROOT_DIR + "/" + snapshotDao.findById(obj.getId()).getAccountId() + "/" + snapshot.getVolumeId());
+                    String snapshotInstallPath = TemplateConstants.DEFAULT_SNAPSHOT_ROOT_DIR + "/" + snapshotDao.findById(obj.getId()).getAccountId() + "/" + snapshot.getVolumeId();
+                    if (snapshot.getHypervisorType().equals(HypervisorType.VMware)) {
+                        snapshotInstallPath = snapshotInstallPath.concat("/" + UUID.randomUUID().toString());
```

Comment:
I would also call contact twice: 1 for the "/"; and the 2nd time for the UUID.randomUUID().toString().